### ISSUE TITLE
FiniteMPS.norm() returns exactly 0.0 for symmetric MPS: _raw_overlap yields 0j

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#819]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Split out of #816, where `FiniteMPS.norm()` returning `0.0` was noted as "possibly related". It is **not** related to #816, and it is **broader than symmetric tensors**.

## Measurements

`manual` is the norm from contracting the MPS explicitly; for the dense row it is confirmed against the full state vector.

| MPS | `norm()` | true norm |
|---|---|---|
| fresh symmetric, **no DMRG** | **0.000000e+00** | 5.153052 |
| after DMRG, fermionic, bases aligned | **0.000000e+00** | 1.000000 |
| after DMRG, spin XXZ | **0.000000e+00** | 1.000000 |
| **dense** `FiniteMPS.random` | **8.005142e-03** | **1.000000** |

## What this rules out

- **Not #816.** The symmetric zero reproduces with the MPS and MPO physical bases correctly aligned, and on a *fresh* MPS that has never been through DMRG.
- **Not fermionic / Jordan-Wigner.** Spin XXZ gives `0.0` too.
- **Not DMRG.** `build_random_symmetric_mps` alone is enough.
- **Not symmetric-only** — this is the part the #816 note did not catch. The dense path is wrong as well, just not conspicuously: it returns `8.005e-03` for a state whose norm is exactly `1.0`, confirmed two ways (`||psi||_2` from the full 2^L vector, and `<psi|psi>`). A wrong-but-plausible number is worse than an obvious zero, because nothing looks broken.

## Minimal reproduction

Three lines, no DMRG, no MPO:

```python
import jax
from tenax.core.mps import FiniteMPS

d = FiniteMPS.random(L=6, d=2, chi=8, key=jax.random.PRNGKey(0))
print(d.norm())     # 0.008005142375507137, but the state has norm 1.0
```

Symmetric variant:

```python
from tenax.core.mps import FiniteMPS
from tenax.algorithms.dmrg import build_random_symmetric_mps

m = build_random_symmetric_mps(L=6, bond_dim=8, seed=1, target_charge=0)
f = FiniteMPS.from_tensors([m.get_tensor(i) for i in range(6)])
print(f.norm())     # exactly 0.0, but the state has norm 5.153052
```

## Where to look

```python
def norm(self) -> float:
    raw = self._raw_overlap(self)
    return float(jnp.exp(self.log_norm) * jnp.sqrt(jnp.abs(raw)))
```

Two candidate faults, and the two failure modes suggest both are in play:

1. `_raw_overlap(self)` yields `0j` on the symmetric path — an exact zero points at a label/flow pairing that matches no blocks, rather than a numerical cancellation.
2. The `log_norm` bookkeeping is wrong on the dense path — the returned value is off by a constant factor (~125x here) rather than being zero, which is the signature of a mis-tracked accumulated scale.

## Why it matters

`norm()` is public API and the natural validity check on a state. In #816 the reporter reached for it to sanity-check a suspicious MPS and could not use it — the check that should have caught a wrong state was itself broken. Any downstream normalization or fidelity computed from `norm()` is currently wrong.

Suggested: a test asserting `norm()` agrees with an explicit contraction for both a dense and a symmetric MPS, since neither backend is currently covered.

Refs #816.

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

**Correction: the dense row in this issue is wrong. There is no dense bug.**

I measured `norm()` against a manual contraction of the site tensors, which ignores `log_norm`. But the documented semantics are

```
|psi> = exp(log_norm) * |psi_tensors>        # class docstring
norm() == exp(log_norm) * sqrt(<tensors|tensors>)
```

For the dense case the split is:

```
log_norm     = -4.827671
raw_overlap  =  1.0          <- tensors are normalized, correctly
norm()       =  exp(-4.8277) * 1 = 8.005e-03
```

`FiniteMPS.random` returns `right_canonicalize()`, which normalizes the tensors and banks the original scale in `log_norm`. So `8.005e-03` **is** the norm of the state that was generated, and `norm()` reports it correctly. My "true norm 1.0" was the norm of the tensors alone — the wrong quantity.

**The symmetric failure is real and unaffected by this correction:**

```
log_norm     = 0.0
raw_overlap  = 0j            <- should be 5.153^2 ~ 26.6
norm()       = 0.0
```

Here `log_norm` is genuinely 0 (the MPS came from `from_tensors`, never canonicalized), so the entire error is in `_raw_overlap` returning an exact zero on the symmetric path. An exact `0j` points at a charge/flow pairing that matches no blocks rather than any numerical cancellation.

So the scope narrows to: **`_raw_overlap` is broken for `SymmetricTensor` MPS.** The "off by ~125x for dense" in the title and table should be disregarded — I will retitle. Everything else stands, including that it is independent of #816, not fermionic, and reproduces on a fresh MPS with no DMRG.
